### PR TITLE
webkitgtk: Fix webkitgtk for macOS

### PR DESCRIPTION
**Pull request by hamishmack**
###### Motivation for this change

The latest webkitgtk in nixpkgs does not currently build on macOS.  This fixes thew build and webkitgtk seems to work correctly on macOS.

I have included PR https://github.com/NixOS/nixpkgs/pull/33304 (fixes geoclue2 on macOS) in this PR as it is also needed for this to work (since webkitgtk uses geoclue2). 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Mic92** [comment]
@GrahamcOfBorg build webkitgtk

**Comment by Mic92** [comment]
cc @LnL7

**Review by matthewbauer — CHANGES_REQUESTED** [review]
Merge conflict.

File: pkgs/development/libraries/glib-networking/default.nix [hunk 1]
```diff
@@ -24,6 +24,8 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [ pkgconfig intltool ];
   propagatedBuildInputs = [ glib gnutls libproxy gsettings_desktop_schemas ];
 
+  NIX_LDFLAGS = stdenv.lib.optionalString stdenv.isDarwin "-lintl";
```
Comment (matthewbauer) [inline pkgs/development/libraries/glib-networking/default.nix]: This should no longer be required.

**Comment by mmahut** [comment]
Are there any updates on this pull request, please?

**Comment by jtojnar** [comment]
Should be resolved by https://github.com/NixOS/nixpkgs/pull/123298 for now.

**Pull request closed without merging**